OctopusDeploy/Sashimi
# Refactor DeployAwsCloudFormationCommand & final cleanup

As title, just to move away from IConvention installation approach (with the minimum effort).

# How to review this PR

- Ensure convention is removed correctly
- Ensure Factory pattern has properly addressed the `AwsEnvironmentGeneration` issue raised early in our Autofac container registration

**Note:** this is not going to master yet and of course not tested yet.
